### PR TITLE
Database url builder

### LINKED ISSUE #25: Add support for MySQL option file
Django supports configuring the database via a MySQL configuration file. If a project using django-sanitized-dump is using this method, extracting the database URL from Django settings will fail.

See https://docs.djangoproject.com/en/2.2/ref/databases/#connecting-to-the-database

### PR DESCRIPTION
**Pull request by JoaRiski**
Add support for mysql options file

Support reading the mysql options file, in case that is being used in
the Django configuration.

Resolves #25

### CODE REVIEW

File: sanitized_dump/utils/db.py [hunk 1]
```diff
@@ -1,41 +1,93 @@
...
+    def create_from_django_database(cls, django_database):
+        engine = get_database_engine_from_django_database(django_database)
+        if not engine:
+            raise ValueError("Database configuration not supported")
+
+        kwargs = {
+            "engine": engine,
+            "user": django_database.get("USER"),
+            "password": django_database.get("PASSWORD"),
+            "host": django_database.get("HOST", None) or "localhost",
```
Comment (frwickst) [inline sanitized_dump/utils/db.py]: Wouldn't just "localhost" instead of "None" suffice here?

Comment (JoaRiski) [reply sanitized_dump/utils/db.py]: Not if someone passes in `None` as the host (like my tests) 😄

File: sanitized_dump/utils/db.py [hunk 2]
```diff
@@ -1,41 +1,93 @@
...
+            "port": django_database.get("PORT", None) or DEFAULT_PORTS.get(engine),
```
Comment (suutari-ai) [inline sanitized_dump/utils/db.py]: Why do you need DEFAULT_PORTS? IMHO if the settings doesn't specify a port, then the database URL shouldn't specify a port either.

Comment (JoaRiski) [reply sanitized_dump/utils/db.py]: Fixed

**Comment by JoaRiski** [comment]
Merging once tests pass, as approval was given over slack
image

**Pull request merged**